### ISSUE TITLE
[grpc-web] ConnectError: add grpc-status code

### ISSUE THREAD

**Issue by jagobagascon** [bufbuild/connect-es#713]
Hi! I'm working on migrating our project from gRPC-Web to connect-es. And I'm having trouble getting the `grpc-status` error code when catching errors.

I'm trying to capture not found gRPC errors but the `ConnectError.code` field is always set to `2 (unknown)`:

```ts
catchError((e) => {
  // e.code is always 2 (unknown)
  if (e instanceof ConnectError && e.code == Code.NotFound) {
    return of(true); // ignore this error
  }
  throw e;
}),
```

I tried debugging the error constructor, and only the first argument of the `ConnectError` is set. The error code is always `2 (unknown)`:
https://github.com/bufbuild/connect-es/blob/0379ca20e05b8ce3ea70f03455e014175e2b7b1d/packages/connect/src/protocol/run-call.ts#L150-L155

Even though the incoming `reason` object includes the `5 (not found)` code:
```js
reason: Error: [not_found] at ...
  code: 5
  message: "[not_found]"
  stack: "Error: [not_found]\n    ..."
  [[Prototype]]: Error
```

**Comment by smaye81** [comment]
Hi @jagobagascon can you elaborate on your environment a bit? For example:

1. Are you using Connect-ES with the Connect protocol or the gRPC-Web protocol?
2. Is the server a gRPC server or is that also Connect? I'm assuming it's gRPC but want to verify.
3. Related to 2, what is the server implementation (i.e. Go, Node, etc.)
4. Is there any intermediary such as Envoy in the path?

**Comment by jagobagascon** [comment]
> Hi @jagobagascon can you elaborate on your environment a bit? For example:
> 
> 1. Are you using Connect-ES with the Connect protocol or the gRPC-Web protocol?

I'm using Connect-ES with the gRPC-Web protocol.
```ts
createGrpcWebTransport({
  baseUrl: s.getApiUrl(),
  interceptors: interceptors,
})
```

> 2. Is the server a gRPC server or is that also Connect? I'm assuming it's gRPC but want to verify.
> 3. Related to 2, what is the server implementation (i.e. Go, Node, etc.)
> 4. Is there any intermediary such as Envoy in the path?

It is a gRPC server implemented in Go. And there's an Envoy Proxy in between.

The HTTP response I'm receiving includes the `Grpc-Status: 5` header.
<img width="196" alt="Screenshot 2023-07-10 at 08 22 29" src="https://github.com/bufbuild/connect-es/assets/12712433/66624875-a3e6-47d8-b2ed-bfd79cc655ec">

But the error thrown does not:
<img width="499" alt="Screenshot 2023-07-10 at 08 20 42" src="https://github.com/bufbuild/connect-es/assets/12712433/a058e95b-2737-40cc-8022-bfb694457496">

**Comment by smaye81** [comment]
OK, I have a suspicion what this might be. Can you confirm the versions you're using for all the `@bufbuild` packages in your package.json?  i.e. `@bufbuild/connect`, `@bufbuild/connect-web`, etc.

**Comment by jagobagascon** [comment]
We're using `@bufbuild/connect-web: 0.10.1` and `@bufbuild/protobuf: 1.2.1`.

**Comment by smaye81** [comment]
OK, can you try this:

```
npm install @bufbuild/connect@latest @bufbuild/connect-web@latest
```

This should give v0.11.0 for both. Once installed, can you try testing again and see if the problem still occurs?

**Comment by jagobagascon** [comment]
The error was still occurring with the latest version. I did some more tests trying to reproduce the issue in a smaller example project, but I wasn't able to make it fail. And after lots of debugging I realised that **our own error interceptor was messing with how the connect-es library handles errors**.

```ts
private errorInterceptor: Interceptor = (invoker) => async (request) => {
  return invoker(request).catch((err) => {
    throw new GrpcErrorResponse(err.code, err.message);
  });
};
```

We are converting gRPC errors to our own error class, and it looks like connect-es is catching it afterwards and converting it back to `ConnectError`, losing the information in the transformation.

I was able to fix it by making our error class extend `ConnectError`, that way the library does not do any conversion. But it forces me to include `@bufbuild/connect` as dependency in a project that does not really need it.

Is this the intended behaviour? I would expect whatever we do on our interceptor to be returned by the services.
